### DISCUSSION TITLE
The underlying Ray payload

### DISCUSSION THREAD

**Discussion by jonathanbossenger** [spatie/ray#432] [Q&A] [answered]
Hi Freek

I have two questions about the inner workings of Ray, for an article I'm writing on debugging tools for PHP.

- Is it correct to say that the Ray object sends a JSON payload to the Ray app, using something similar to when a developer makes an API request using an HTTP client, like Guzzle?
- Is it correct to say that the Ray::Pause functionality initiates a PHP `sleep` function call, which will keep running until the code is unpaused from the Ray app.

Thanks in advance.

**Comment by freekmurze** [comment, marked as answer]
> Is it correct to say that the Ray object sends a JSON payload to the Ray app, using something similar to when a developer makes an API request using an HTTP client, like Guzzle?

yup!

> Is it correct to say that the Ray::Pause functionality initiates a PHP sleep function call, which will keep running until the code is unpaused from the Ray app.

Yup, that's correct.
